src/open_feature.py

```diff
@@ -0,0 +1,20 @@
+from src.provider.provider import AbstractProvider
+
+global provider
+
+
+class OpenFeature:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def set_provider(provider_type: AbstractProvider):
+        if provider_type is None:
+            print("No provider")
```

Comment:
Let's raise an exception here rather than print. Probably `TypeError` is the most appropriate.

Reply:
TypeError raised now instead